### ISSUE TITLE
Deployment does not set production mode

### ISSUE THREAD

**Issue by dverkade** [davidalger/capistrano-magento2#68]
Hi,

The capistrano tasks does not set the production mode. Is it possible to add a step to set the deployment to the production mode automatically?

**Comment by davidalger** [comment]
@dverkade Thanks for writing! This is intentional behavior, as not to limit the scenarios in which the gem may be used.

The deploy mode is a setting which M2 stores in the app/etc/env.php file, or it can also be overridden via an environment variable set in the nginx configuration (for example). This means any setting of production mode we would do would also end up being unreliable.

My recommendation would be to ensure that the following is in your env.php file, or otherwise set via an env variable in the server config:

      'MAGE_MODE' => 'production'

**Comment by dverkade** [comment]
Hi David,

Thanks for your feedback! I'll discuss this with my team and see how we are going to implement this.

**Comment by davidalger** [comment]
@dverkade No problem! If you use nginx, and go down the route of putting it in server configuration, this is the line we have in the application nginx config we deploy on over here at Classy Llama:

   set $MAGE_MODE production;

**Comment by davidalger** [comment]
@dverkade No problem! If you use nginx and go down the route of putting it in the server configuration, this is what we include in our application nginx config at Classy Llama:

This line at top of nginx app config file:

    set $MAGE_MODE production;

Then we include this in the `location` with the `fastcgi_pass` directive:

    fastcgi_param MAGE_MODE $MAGE_MODE;

**Comment by davidalger** [comment]
Update here, I've added tasks to show current mode (as defined in env.php) and to set production mode. As of v0.7.0 production mode is set during deployment prior to static content deployment. Honestly the decision was both because I had remembered you'd requested such and also because 2.2.0-rc introduces a check into static content deployment which fails the deploy if production mode is not set (unless -f is used, although I'm still specifying -f for 2.2 deploy just in case MAGE_MODE is not set to production in the env for the shell).
